src/v/utils/object_pool.h

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright 2022 Redpanda Data, Inc.
+ *
+ * Licensed as a Redpanda Enterprise file under the Redpanda Community
+ * License (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * https://github.com/redpanda-data/redpanda/blob/master/licenses/rcl.md
+ */
+
+#pragma once
+
+#include "seastarx.h"
+
+#include <seastar/core/abort_source.hh>
+#include <seastar/core/future.hh>
+
+#include <concepts>
+#include <deque>
+#include <stack>
+
+template<std::movable T>
+class object_pool {
```

Comment:
Utility classes should especially have a class comment & documented API.

Reply:
Adding documentation to the class and its public functions.